aspnetcore/blazor/file-uploads.md

```diff
@@ -74,21 +77,295 @@ The following example demonstrates multiple image file upload in a component. `I
+
+    public int ErrorCode { get; set; }
+}
+```
+
+`Pages/UploadFiles.razor` in the **`Client`** project of a hosted Blazor solution:
+
+```razor
+@page "/upload-files"
+@using System.IO
+@using System.Linq
+@using System.Net
+@using Microsoft.Extensions.Logging
+@using FileUploadTesting.Shared
+@inject HttpClient Http
+@inject ILogger<UploadFiles> logger
+
+<h1>Upload files</h1>
+
+<p>
+    Upload one or more PNG image files:
+    <InputFile OnChange="@OnInputFileChange" multiple />
+</p>
+
+@if (images.Count > 0)
+{
+    <div class="card">
+        <div class="card-body">
+            @foreach (var image in images)
+            {
+                <figure>
+                    <img class="rounded m-1" src="@image.Thumbnail" />
+                    <figcaption>
+                        @WebUtility.HtmlEncode(image.Name)
+                        <br>
+                        @if (FileUpload(uploadResults, image.Name, logger,
+                            out var result))
+                        {
+                            @result.StoredFileName
+                        }
+                        else
+                        {
+                            <span>
+                                There was an error uploading the file
+                                (Error: @result.ErrorCode).
+                            </span>
+                        }
+                    </figcaption>
+                </figure>
+            }
+        </div>
+    </div>
+}
+
+@code {
+    private IList<Image> images = new List<Image>();
+    private IList<UploadResult> uploadResults =
+        new List<UploadResult>();
+
+    private async Task OnInputFileChange(InputFileChangeEventArgs e)
+    {
+        var format = "image/png";
+        var maxAllowedFiles = 3;
+        var maxWidth = 640;
+        var maxHeight = 480;
+        var thumbnailSize = 100;
+        long maxFileSize = 1024 * 1024 * 15;
+        var upload = false;
+
+        using var content = new MultipartFormDataContent();
+
+        foreach (var imageFile in e.GetMultipleFiles(maxAllowedFiles))
+        {
+            if (uploadResults.SingleOrDefault(
+                f => f.FileName == imageFile.Name) is null)
+            {
+                var image = new Image() { Name = imageFile.Name };
+
+                var resizedImageFile = await imageFile.RequestImageFileAsync(
+                    format, thumbnailSize, thumbnailSize);
+                var buffer = new byte[resizedImageFile.Size];
+                await resizedImageFile.OpenReadStream().ReadAsync(buffer);
+                image.Thumbnail =
+                    $"data:{format};base64,{Convert.ToBase64String(buffer)}";
+
+                var uploadFile = await imageFile.RequestImageFileAsync(
+                    format, maxWith: maxWidth, maxHeight: maxHeight);
+                using Stream s = uploadFile.OpenReadStream(maxFileSize);
+                using MemoryStream ms = new();
+                await s.CopyToAsync(ms);
```

Comment:
Any reason there's two separate patterns for reading the stream? The thumbnail reads in to a byte-array, this one reads in to memorystream -> ToArray? Maybe this could use a `StreamContent` instead and that way you don't have to read it at all?